toml/src/main/kotlin/org/rust/toml/search/CargoTomlUsageTypeProvider.kt

```diff
@@ -0,0 +1,41 @@
+/*
+ * Use of this source code is governed by the MIT license that can be
+ * found in the LICENSE file.
+ */
+
+package org.rust.toml.search
+
+import com.intellij.psi.PsiElement
+import com.intellij.usages.UsageTarget
+import com.intellij.usages.impl.rules.UsageType
+import com.intellij.usages.impl.rules.UsageTypeProviderEx
+import org.rust.lang.core.RsPsiPattern.anyCfgFeature
+import org.rust.lang.core.psi.RsLitExpr
+import org.rust.toml.CargoTomlPsiPattern.onFeatureDependencyLiteral
+import org.toml.lang.psi.TomlLiteral
+
+object CargoTomlUsageTypeProvider : UsageTypeProviderEx {
+    private val FEATURE_DEPENDENCY = UsageType { "Cargo feature dependency" }
+    private val DEPENDENCY_FEATURE = UsageType { "Package dependency" }
+    private val CFG_FEATURE = UsageType { "Cfg attribute" }
```

Comment:
Let's move them into companion object after converting `object` to general class